release.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -e
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+get_latest() {
+  git fetch
+
+  git tag |
+    # drop `v` prefix
+    grep "^v" |
+    cut -c2- |
+
+    # sort by semantic version
+    sort -t "." -k1,1n -k2,2n -k3,3n |
+
+    # last
+    tail -n 1 |
+
+    # drop newline
+    tr -d '\n'
+}
+
+NEW="$1"
+
+if [ -z "$NEW" ]; then
+  echo "Usage  : bash release.sh <version>"
+  echo "Example: bash release.sh 1.2.3"
+  echo ""
+  echo "Fetching tags..."
+  latest="$(get_latest)"
+  echo -n "The current version is: $latest"
+
+  exit 1
+fi
+
+if [[ "$NEW" == v* ]]; then
+  echo "<version> must not start with \"v\""
+  exit 1
+fi
+
+echo "Checking for clean working tree..."
+if [[ "$(git diff --stat)" != "" ]]; then
+  echo "❌ Dirty working tree (try git stash)"
```

Comment:
Emojis always looks funky as heck to me in bash scripts 🤷

Reply:
I like when they're used sparingly 😄